bezier-rs/lib/src/lib.rs

```diff
@@ -724,13 +745,55 @@ impl Bezier {
 				.collect::<Vec<f64>>();
 		}
 
-		// If the self is linear, then use the implementation for intersections with linear lines
-		if self.handles == BezierHandles::Linear {
-			return curve.intersections(self, Some(error));
+		// TODO: Consider using the `intersections_between_vectors_of_curves` helper function here
+		// Otherwise, use bounding box to determine intersections
+		self.intersections_between_subcurves(0. ..1., other, 0. ..1., error).iter().map(|t_values| t_values[0]).collect()
+	}
+
+	/// Helper function to compute intersections between lists of subcurves.
+	/// This function uses the algorithm implemented in `intersections_between_subcurves`.
+	fn intersections_between_vectors_of_curves(subcurves1: &[(Bezier, Range<f64>)], subcurves2: &[(Bezier, Range<f64>)], error: f64) -> Vec<[f64; 2]> {
+		let segment_pairs = subcurves1.iter().flat_map(move |(curve1, curve1_t_pair)| {
+			subcurves2.iter().filter_map(move |(curve2, curve2_t_pair)| {
+				if utils::do_rectangles_overlap(curve1.bounding_box(), curve2.bounding_box()) {
+					Some((curve1, curve1_t_pair, curve2, curve2_t_pair))
+				} else {
+					None
+				}
```

Comment:
It would be slightly more idiomatic to use `.then()` here, see https://stackoverflow.com/questions/54841351/how-do-i-idiomatically-convert-a-bool-to-an-option-or-result-in-rust